programs/test/query_compile_time_config.c

```diff
@@ -28,20 +28,27 @@
 #define MBEDTLS_EXIT_FAILURE EXIT_FAILURE
 #endif
 
-#define USAGE                                                                \
-    "usage: %s [ <MBEDTLS_CONFIG> | -l ]\n\n"                                \
-    "This program takes one command line argument which corresponds to\n"    \
-    "the string representation of a Mbed TLS compile time configuration.\n"  \
-    "The value 0 will be returned if this configuration is defined in the\n" \
-    "Mbed TLS build and the macro expansion of that configuration will be\n" \
-    "printed (if any). Otherwise, 1 will be returned.\n"                     \
-    "-l\tPrint all available configuration.\n"
+#define USAGE                                                                   \
+    "usage: %s [ -all | -any | -l ] <MBEDTLS_CONFIG> ...\n\n"                   \
+    "This program takes command line arguments which correspond to\n"           \
```

Comment:
This wording is a bit unclear. I suggest replacing lines 35..38 with something like:

If "--all" and "--any" are not used, then, if all given arguments are defined in the Mbed TLS build, 0 is returned; otherwise 1 is returned. If only one argument is given, the macro expansion of that configuration will be printed (if any).